Tintanaa/rag_front
# Feat1 prerequisities

## Merge Request: Basic Theming and Splash Screen Implementation

This merge request adds basic color schemes, dark and light themes beginning setups, path to assets and a simple splash screen to the Flutter project.

Changes:

* `assets/`:
    * Introduced folder for storing assets needed for project
    * There are 3 folders: 
        * images for .jpg, .png formats
        * vectors for .svg formats
        * fonts for custom .ttf fonts (if they will be needed)
* `lib/src/features/splash/splash.dart`:
    * Implemented splash screen as StatelessWidget with logo on it.
* `lib/src/localization/`
    * Implemented layer for storing localization related files in .arb format. For now only Russian language is supported
* `lib/src/theme`:
    * Implemented layer for storing information about color palettes, text styles, application themes. 
    * Defined color palettes for light and dark themes.
    * Created theme data for both light and dark themes.
    * Added text styles
* `pubspec.yaml`:
    * Added dependencies for carousel_slider, flutter_svg

To test these changes:

1. Run the application.
2. Observe the splash screen on application launch.
3. Switch between light and dark themes by modifying the `RagTheme` in the main.dart
